pkg/action/install.go

```diff
@@ -584,14 +592,107 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
 		}
 	}
 
-	if pr != nil {
-		b, err = pr.Run(b)
+	return hs, b, notes, nil
+}
+
+// postProcessFiles sends files for post-processing.
+// Some post-processing may need all objects at once.
+// Therefore we concatenate all rendered files, post-process and split them back.
+// A temporary annotation is added to preserve template filename.
+func (c *Configuration) postProcessFiles(files map[string]string, pr postrender.PostRenderer) (map[string]string, error) {
+	if pr == nil {
+		return files, nil
+	}
+
+	annotationName := "helmPostProcessFileName"
+
+	b := bytes.NewBuffer(nil)
+
+	// Split the rendered files into documents and add the temp annotation.
+	reYamlDocumentSeparator := regexp.MustCompile(`\n---\s*\n`)
+	for fileName, fileContent := range files {
+		for _, document := range reYamlDocumentSeparator.Split(fileContent, -1) {
+			if strings.TrimSpace(document) == "" {
+				continue
+			}
+			b.WriteString("---\n")
+			documentYaml, _, err := yamlSetTemporaryAnnotation(document, annotationName, fileName)
+			if err != nil {
+				return nil, fmt.Errorf("cannot prepare a file for post-process: %w", err)
+			}
+			b.Write(documentYaml)
+		}
+	}
+
+	// Run post-processing.
+	b, err := pr.Run(b)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while running post render on files")
+	}
+
+	postProcessedFiles := make(map[string]string)
+	// Split the post-processed stream into files.
+	for _, document := range strings.Split(b.String(), "---\n") {
```

Comment:
Should this be using the regex from above to split?

Reply:
I though initially that these document separators are guaranteed to be `"---\n"`, because they are added in the same function a few lines above. So I thought `strings.Split()` would save a few CPU cycles compared to `regexp.Split()`.
But now when you asked, it is indeed possible that post-renderer adds own separators that may have whitespaces, which may indent (and invalidate) the line after separator in case a simple `strings.Split()` is used. You're right, `regexp.Split()` must be used there.